### PR TITLE
[FA] First draft of Chapter4/Page2

### PR DESCRIPTION
**Pull request by hamedonline**
- added Persian translation for chapter4/page2
- updated _toctree.yml to include translated page title
- updated glossary page with new word proposals

Related to https://github.com/huggingface/course/issues/50

Would appreciate your valuable review dear @jowharshamshiri

### CODE REVIEW

File: chapters/fa/chapter4/2.mdx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+هاب مدل، انتخاب مدل مناسب را تسهیل می‌بخشد؛ به طوری که می‌توان با چند خط کد، آن را در هر کتابخانه پایین‌دستی وابسته استفاده نمود. بیایید به نحوه عملی به کارگیری یکی از این مدل‌ها نگاهی انداخته و ببینیم چگونه می‌توانیم در مقام کمک به جامعه کاربران برآییم.
+
+فرض کنید به دنبال مدلی مبتنی بر زبان فرانسوی هستیم که قادر به پر کردن جاهای خالی متن است.
+
+<div class="flex justify-center">
+<img src="https://huggingface.co/datasets/huggingface-course/documentation-images/resolve/main/en/chapter4/camembert.gif" alt="انتخاب مدل Camembert" width="80%"/>
+</div>
+
+نقطه تعلیم `camembert-base` را انتخاب می‌کنیم تا مدل را با آن آزمایش نماییم. برای شروع استفاده از آن، تمام آنچه نیاز داریم شناسه `camembert-base`  است! همان گونه که در فصل‌های پیشین دیدید، می‌توانیم با استفاده از تابع `pipeline()` نمونه‌ای از آن بسازیم.
```
Comment (jowharshamshiri) [inline chapters/fa/chapter4/2.mdx:34]: T/point13 Please put function name in a span.

File: chapters/fa/chapter4/2.mdx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  {'sequence': 'Le camembert est excellent :)', 'score': 0.1055697426199913, 'token': 2183, 'token_str': 'excellent'}, 
+  {'sequence': 'Le camembert est succulent :)', 'score': 0.03453313186764717, 'token': 26202, 'token_str': 'succulent'}, 
+  {'sequence': 'Le camembert est meilleur :)', 'score': 0.0330314114689827, 'token': 528, 'token_str': 'meilleur'}, 
+  {'sequence': 'Le camembert est parfait :)', 'score': 0.03007650189101696, 'token': 1654, 'token_str': 'parfait'}
+]
+```
+
+</div>
+
+همان طور که می‌بینید، بارگذاری مدل درون خط تولید بسیار ساده است. تنها چیزی که باید مراقبش باشید این است که نقطه تعلیم انتخاب شده، مناسب وظیفه‌ای باشد که برای انجام آن به کار گرفته خواهد شد. برای مثال، در اینجا نقطه تعلیم `camembert-base` را در خط تولید `fill-mask` استفاده می‌کنیم، که انتخابی کاملا معقول است. اما اگر ما این نقطه تعلیم را در خط تولید `text-classification` استفاده کنیم، نتایج هیچ معنایی نخواهند داشت، زیرا سَر مربوط به `camembert-base`، برای این کار مناسب نیست. توصیه می‌کنیم که از انتخاب‌گر وظیفه در رابط کاربری هاب هاگینگ‌فِیس برای انتخاب نقاط تعلیم مناسب استفاده نمایید:
```
Comment (jowharshamshiri) [inline chapters/fa/chapter4/2.mdx:57]: We previously used مسئله for task in multiple chapters. Somehow that's not made it to G. Please add to G and use same here. i. e. انتخاب‌گر نوع مسئله

File: chapters/fa/chapter4/2.mdx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+```py
+from transformers import CamembertTokenizer, CamembertForMaskedLM
+
+tokenizer = CamembertTokenizer.from_pretrained("camembert-base")
+model = CamembertForMaskedLM.from_pretrained("camembert-base")
+```
+
+</div>
+
+با این حال، توصیه می‌کنیم به جای این کار از [`Auto*` classes](https://huggingface.co/transformers/model_doc/auto.html?highlight=auto#auto-classes) استفاده کنید؛ چرا که طراحی این کلاس‌های خودکار به گونه‌ایست که فارغ از هرگونه وابستگی به معماری به کار رفته هستند. در حالی که نمونه کد قبلی، کاربران را محدود به نقاط تعلیم قابل بارگذاری در معماری CamemBERT می‌کند، استفاده از کلاس‌های `Auto*`، سهولت تعویض نقاط تعلیم را در بر دارد:
```
Comment (jowharshamshiri) [inline chapters/fa/chapter4/2.mdx:78]: I propose the following for "`Auto*` classes":
کلاس‌هایی که نام‌‌ آنها با Auto شروع می‌شود.

File: chapters/fa/glossary/1.mdx [hunk 4]
```diff
@@ -110,20 +114,21 @@
...
 | Prediction, as in nn model | پیش‌بینی |
+| Bias | سوگیری |
 | Logit, as in math and also in Pytorch | لوجیت؟ |
 | SoftMax | سافت‌مکس |
 | Loss Function | تابع هزینه |
 | Activation Layer | لایه فعال‌سازی |
 | Cross Entropy | آنتروپی متقابل |
 | Head, as in model | سَر |
 | Weight, as in model | وزن |
-| Weights, as in model | وزون؟ |
```
Comment (jowharshamshiri) [inline chapters/fa/glossary/1.mdx:125]: Please include the equivalent you suggested for the plural form to avoid confusion in future for other collaborators.

File: chapters/fa/chapter4/2.mdx (hunk 2 above)
Comment (jowharshamshiri) [reply chapters/fa/chapter4/2.mdx:57]: Also what do Persian-speaking UI/UX people use for "selector"? انتخاب‌گر seems a little weird.

File: chapters/fa/_toctree.yml [hunk 5]
```diff
@@ -8,7 +8,7 @@
   - local: chapter1/1
     title: مقدمه
 
-- title: ۲- بکارگیری ترنسفورمرهای هاگینگ‌فِیس
```
Comment (jowharshamshiri) [inline chapters/fa/_toctree.yml:11]: I think بکارگیری is one word. This case is different from cases we discussed before.

**Review by jowharshamshiri — CHANGES_REQUESTED** [review]
Great Job. ❤️ 
I suggest doing a second draft later on. My feedback on this draft is:
-Needs to sound more natural to a Persian-speaker(less English-like sentence compositions. I can be guilty of this myself.)
-Needs to sound less literary(بر مقام فلان برآییم and the like don't sound like they belong in a technical CS text with a friendly, inviting tone).

File: chapters/fa/chapter4/2.mdx (hunk 2 above)
Comment (hamedonline) [reply chapters/fa/chapter4/2.mdx:57]: > Also what do Persian-speaking UI/UX people use for "selector"? انتخاب‌گر seems a little weird.

Well, 'انتخاب‌گر' is the closest Persian match I could relate to, open for better suggestions ofc. The alternative option would be to ignore literal translation for this sentence and go with an equivalent like:
"توصیه می‌کنیم برای گزینش نقاط تعلیم مناسب، از قسمت انتخاب مسئله در رابط کاربری هاب هاگینگ‌فِیس استفاده نمایید:"
Please let me know which one looks more conventional to you.

Comment (jowharshamshiri) [reply chapters/fa/chapter4/2.mdx]: Very cool and creative wording you're using here. I'm super cool with it. Just asked a UI/UX friend what they use day-to-day. Will update you when/if they get back to me with a better alternative.

**Comment by hamedonline** [comment]
> Great Job. ❤️ I suggest doing a second draft later on. My feedback on this draft is: -Needs to sound more natural to a Persian-speaker(less English-like sentence compositions. I can be guilty of this myself.) -Needs to sound less literary(بر مقام فلان برآییم and the like don't sound like they belong in a technical CS text with a friendly, inviting tone).

Dear @jowharshamshiri thanks for the review ❤️, your views and points are applied. I also tried to change the overall tone by modifying some sentences, hope it rhymes more technical now. Please check them back after the build and if it’s ok with you, we can go ahead with the merge.
I forgot some modifications to TF part, will apply them later today.

**Comment by jowharshamshiri** [comment]
> > Great Job. ❤️ I suggest doing a second draft later on. My feedback on this draft is: -Needs to sound more natural to a Persian-speaker(less English-like sentence compositions. I can be guilty of this myself.) -Needs to sound less literary(بر مقام فلان برآییم and the like don't sound like they belong in a technical CS text with a friendly, inviting tone).
> 
> Dear @jowharshamshiri thanks for the review ❤️, your views and points are applied. I also tried to change the overall tone by modifying some sentences, hope it rhymes more technical now. Please check them back after the build and if it’s ok with you, we can go ahead with the merge. I forgot some modifications to TF part, will apply them later today.

Thanks Hamed jan, very kind of you. And for the record when I said maybe do another draft later on, I didn't mean to hold up this PR. In any case thanks for doing it right here in this one. Great job. I don't know what TF part you're talking about, don't really have the context in my head right now, am cool with whatever you decide to do. ❤️

**Review by jowharshamshiri — APPROVED** [review]
LGTM 😍

**Comment by lewtun** [comment]
Thank you for the translation @hamedonline and @jowharshamshiri for the review 🚀 ! 

I've checked this section renders on the PR build, so let me know when you'd like me to merge this :)

File: chapters/fa/chapter4/2.mdx (hunk 2 above)
Comment (hamedonline) [reply chapters/fa/chapter4/2.mdx]: > UPDATE: @hamedonline jan, apparently they say منوی دراپ داون in day-to-day use, but they've also heard منوی کشویی. In any case, I like your wording much better, but if we need a one-word equivalent some where else we should probably use one of these.

Great, thanks for the update. Sure, later on we'll put those equivalents to use under different circumstances.

**Comment by hamedonline** [comment]
> Thanks Hamed jan, very kind of you. And for the record when I said maybe do another draft later on, I didn't mean to hold up this PR. In any case thanks for doing it right here in this one. Great job. I don't know what TF part you're talking about, don't really have the context in my head right now, am cool with whatever you decide to do. ❤️
> 
> P.S. if you do make more changes, add that plural of "weight" to G too.

Sure thing @jowharshamshiri jan, thanks a lot for the feedback ❤️.
Just fixed the translation for TensorFlow part. Plural of "weight" has already been added to G in previous commits, thanks again for reminding.
Dear @lewtun we're good to go for merging 🤩❤️.
